library/core/src/iter/range.rs

```diff
@@ -201,24 +201,12 @@ macro_rules! step_identical_methods {
 
         #[inline]
         fn forward(start: Self, n: usize) -> Self {
-            // In debug builds, trigger a panic on overflow.
-            // This should optimize completely out in release builds.
-            if Self::forward_checked(start, n).is_none() {
-                let _ = Add::add(Self::MAX, 1);
-            }
-            // Do wrapping math to allow e.g. `Step::forward(-128i8, 255)`.
-            start.wrapping_add(n as Self)
+            start + (n as Self)
```

Comment:
Just a gentle reminder for https://github.com/rust-lang/rust/pull/75495#discussion_r470234606